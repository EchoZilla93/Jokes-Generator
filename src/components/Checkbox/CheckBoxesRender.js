import React, {useState} from 'react';
import {checkboxes} from '../../helpers/Helpers';

import {CheckBoxBluePrint} from './CheckBoxBluePrint';
import './checkBoxStyles.scss'

const CheckBoxRender = () => {
    const [selectedId, setSelectedId] = useState(1);
    const changeHandler = (id) => {
        setSelectedId(id);
    };
    
    return(
        <div>
            <div className="checkboxContainer">
                {checkboxes.map(checkBox => (
                    <CheckBoxBluePrint
                        key={checkBox.id}
                        type= {'checkbox'}
                        onSelect = {changeHandler}
                        selectedId = {selectedId}
                        {...checkBox}
                    />
                ))}
            </div>
        </div>
    )
}

export default CheckBoxRender;