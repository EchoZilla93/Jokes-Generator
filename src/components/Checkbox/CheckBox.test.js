import React from 'react';
import CheckBoxesRender from './CheckBoxesRender';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const setUp = (props={}) => {
    const component = shallow(<CheckBoxesRender {...props}/>);
    return component;
}

describe('CheckBoxBluePrint component',() => {
    let component;
    beforeEach(()=> {
        component = setUp();
    })
    it('renders expected snapshot', () => {
        const wrapper = component;
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})