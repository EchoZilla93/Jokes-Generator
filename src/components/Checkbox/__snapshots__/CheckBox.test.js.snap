// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxBluePrint component renders expected snapshot 1`] = `
<div>
  <div
    className="checkboxContainer"
  >
    <CheckBoxBluePrint
      className="checkboxContainer__form-item"
      id={1}
      key="1"
      onSelect={[Function]}
      selectedId={1}
      type="radio"
      value="Random"
    />
    <CheckBoxBluePrint
      className="checkboxContainer__form-item"
      id={2}
      key="2"
      onSelect={[Function]}
      selectedId={1}
      type="radio"
      value="Category"
    />
    <CheckBoxBluePrint
      className="checkboxContainer__form-item"
      id={3}
      key="3"
      onSelect={[Function]}
      selectedId={1}
      type="radio"
      value="Search"
    />
  </div>
</div>
`;
